File: src/main/java/sage/transfer/BlogPreview.java
package sage.transfer;

import java.util.Date;
import java.util.List;

import sage.entity.Blog;
import sage.util.Colls;
import sage.util.Strings;

public class BlogPreview {
  public long id;
  public String title;
  public UserLabel author;
  public String summary;
  public Date createdTime;
  public Date modifiedTime;
  public List<TagLabel> tags;

  BlogPreview() {}

  public BlogPreview(Blog blog) {
    id = blog.getId();
    title = blog.getTitle();
    author = new UserLabel(blog.getAuthor());
    summary = Strings.cut(blog.getContent(), 0, 100);
    createdTime = blog.getWhenCreated();
    modifiedTime = blog.getWhenModified();
    tags = Colls.map(blog.getTags(), TagLabel::new);
  }
}

File: src/main/kotlin/sage/transfer/BlogPreview.kt
package sage.transfer

import sage.annotation.KotlinNoArg
import sage.entity.Blog
import sage.util.Strings
import java.sql.Timestamp

@KotlinNoArg
class BlogPreview {
  var id: Long = 0
  var author: UserLabel? = null
  var title: String = ""
  var summary: String = ""
  var whenCreated: Timestamp? = null
  var whenEdited: Timestamp? = null
  var tags: List<TagLabel> = arrayListOf()
  var tweetId: Long = 0
  var commentCount: Int = 0
  var likes: Int = 0

  constructor(blog: Blog) {
    id = blog.id
    author = blog.author.toUserLabel()

    title = blog.title
    if (blog.contentType == Blog.MARKDOWN) {
      summary = Strings.escapeHtmlTag(Strings.omit(blog.inputContent, 103))
    } else {
      summary = Strings.escapeHtmlTag(Strings.omit(blog.content, 103))
    }
    whenCreated = blog.whenCreated
    whenEdited = actualWhenEdited(blog.whenCreated, blog.whenEdited)

    tags = blog.tags.map { it.toTagLabel() }
    tweetId = blog.tweetId
    blog.stat()?.let {
      commentCount = it.comments
      likes = it.likes
    }
  }
}
